src/meta/src/hummock/compaction/picker/min_overlap_compaction_picker.rs

```diff
@@ -56,37 +56,53 @@ impl MinOverlappingPicker {
         level_handlers: &[LevelHandler],
     ) -> (Vec<SstableInfo>, Vec<SstableInfo>) {
         let mut scores = vec![];
-        for left in 0..select_tables.len() {
-            if level_handlers[self.level].is_pending_compact(&select_tables[left].sst_id) {
+        let mut select_file_ranges = vec![];
+        for (idx, sst) in select_tables.iter().enumerate() {
+            if level_handlers[self.level].is_pending_compact(&sst.sst_id) {
                 continue;
             }
             let mut overlap_info = self.overlap_strategy.create_overlap_info();
-            let mut select_file_size = 0;
-            for (right, table) in select_tables.iter().enumerate().skip(left) {
-                if level_handlers[self.level].is_pending_compact(&table.sst_id) {
+            overlap_info.update(sst);
+            let overlap_files_range = overlap_info.check_multiple_overlap(target_tables);
+
+            if overlap_files_range.is_empty() {
+                return (vec![sst.clone()], vec![]);
+            }
+            select_file_ranges.push((idx, overlap_files_range));
+        }
+        select_file_ranges.retain(|(_, range)| {
+            let mut pending_compact = false;
+            for other in &target_tables[range.clone()] {
+                if level_handlers[self.target_level].is_pending_compact(&other.sst_id) {
+                    pending_compact = true;
                     break;
                 }
-                if select_file_size > self.max_select_bytes {
+            }
+            !pending_compact
+        });
+        for left in 0..select_file_ranges.len() {
+            let mut select_file_size = 0;
+            let mut target_level_overlap_range = select_file_ranges[left].1.clone();
+            let mut total_file_size = 0;
+            for other in &target_tables[target_level_overlap_range.clone()] {
```

Comment:
tips: return the `total_file_size` when calling `check_multiple_overlap` to avoid the double calculation

Reply:
No. It is not necessary to record total file size for those pending compact file.